### PR TITLE
adding some logic and design

### PR DESCRIPTION
**Pull request by Frickazoid**

### CODE REVIEW

File: src/disign.css [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+.MoveBtn{
+    width: 100%;
+    border-radius: 15px;
+}
+
+.MoveBtn:hover {
+    background-color: #4CAF50; /* Зеленый */
+    color: white;
+    /*box-shadow: 0 12px 16px 0 rgba(0,0,0,0.24), 0 17px 50px 0 rgba(0,0,0,0.19);*/
```
Comment (Alex-Fishman) [inline src/disign.css]: delete

File: src/disign.css [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+.Flex-container{
+    display: flex;
+    flex-direction: row;
+    /*align-items: center;*/
+    /*align-content: stretch;*/
```
Comment (Alex-Fishman) [inline src/disign.css]: delete

File: src/disign.css [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    margin-bottom: 1em; /* Расстояние снизу */
+}
+
+.UserInfoRight {
+    border-radius: 15px;
+    /*border: 2px solid #73AD21;*/
+    text-align: left;
+    padding-left: 15px;
+    padding-right: 15px;
+    margin-bottom: 1em; /* Расстояние снизу */
```
Comment (Alex-Fishman) [inline src/disign.css]: Using 'em' and 'px' at the same time looks weird.
Please choose one of them, 
Do you understand the difference between "em"/"px"?
"em" is a relative size, while "px" is an absolute size.
https://www.w3schools.com/cssref/css_units.php

File: src/components/ObjectInfo.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+function ObjectInfo({arrangement, onItemPress, side, style}) {
+
+    function onPress({idbtn, gender:gender }) {
```
Comment (Alex-Fishman) [inline src/components/ObjectInfo.js]: `({idbtn, gender:gender })` should be `({idbtn, gender })`

File: src/components/ObjectInfo.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        onItemPress({side, id:idbtn, gender:gender});
```
Comment (Alex-Fishman) [inline src/components/ObjectInfo.js:4]: ({idbtn, gender:gender }) should be ({idbtn, gender })

File: src/components/ObjectInfo.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    return (
+        <li className={'UserInfo'+side} style={style}>
```
Comment (Alex-Fishman) [inline src/components/ObjectInfo.js:8]: `'UserInfo'+side` works but a bad way to write
please use 
`<li className={`UserInfo${side}`}`

Comment (Alex-Fishman) [inline src/components/ObjectInfo.js]: `'UserInfo'+side` works, but it is an old and ugly way to combine strings.  please use this 
```suggestion
        <li className={`UserInfo${side}`} style={style}>
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: src/components/ObjectInfo.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            <div className="Flex-container">
+                <div className="Flex-info-container">
+                    <div className="UserFName">
```
Comment (Alex-Fishman) [inline src/components/ObjectInfo.js]: image

https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

```suggestion
                    <div className="user-f-name">
```

File: src/components/ObjectInfo.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
+function ObjectInfo({arrangement, onItemPress, side, style}) {
+
+    function onPress({idbtn, gender:gender }) {
+        onItemPress({side, id:idbtn, gender:gender});
+    }
+
+    return (
+        <li className={'UserInfo'+side} style={style}>
+            <div className="Flex-container">
+                <div className="Flex-info-container">
```
Comment (Alex-Fishman) [inline src/components/ObjectInfo.js]:
```suggestion
                <div className="flex-info-container">
```

File: src/App.js [hunk 9]
```diff
@@ -1,12 +1,75 @@
...
+import {useEffect, useState} from 'react';
+import './disign.css';
+import ObjectInfo from "./components/ObjectInfo";
 
 function App() {
-  return (
-    <div className="App">
-        <div> hello </div>
+
+    const [leftarr, setlarr] = useState([]);
```
Comment (Alex-Fishman) [inline src/App.js]:
```suggestion
    const [leftArr, setlArr] = useState([]);
```

File: src/App.js [hunk 10]
```diff
@@ -1,12 +1,75 @@
...
+    const [rightarr, setrarr] = useState([]);
```
Comment (Alex-Fishman) [inline src/App.js]:
```suggestion
    const [rightArr, setrArr] = useState([]);
```

File: src/App.js [hunk 11]
```diff
@@ -1,12 +1,75 @@
...
+    const [ backgroundColor, setBackgroundColor ] = useState('#50e3e3');
+
+    useEffect(() => {
+        const dataJSON = require('./MOCK_DATA.json');
+        setlarr(dataJSON);
+        console.log(leftarr.length);
```
Comment (Alex-Fishman) [inline src/App.js]: remove

File: src/App.js [hunk 12]
```diff
@@ -1,12 +1,75 @@
...
+        }
+    }
+
+    function conditionCheck(gender) {
+        let checkState = true;
+        if (gender === 'Male' || gender === 'Female') {
+            const checkArr = rightarr.filter(item => item.gender === gender);
+            if(checkArr.length===5){checkState = false}
+        }
+        return checkState
```
Comment (Alex-Fishman) [inline src/App.js]:
```suggestion
        if (gender === 'Male' || gender === 'Female') {
            const checkArr = rightarr.filter(item => item.gender === gender);
           return checkArr.length !== 5;
        }
        return true
```

checkArr.length !== 5; already returns boolean

File: src/App.js [hunk 13]
```diff
@@ -1,12 +1,75 @@
...
+    }, [rightarr])
+
+    function onItemPress({side, id, gender}) {
+        if (side === 'Left') {
+            if (conditionCheck(gender)) {
+                const from = {arr: leftarr, setArr: setlarr};
+                const to = {arr: rightarr, setArr: setrarr};
+                moveElement({id, from, to});
+            } else {
+                alert('Max 5 male and 5 female on right side');
```
Comment (Alex-Fishman) [inline src/App.js]: i want a better validation.
i want to know what failed right now, female? or male?

**Review by Alex-Fishman — CHANGES_REQUESTED** [review]
First of all good job 👏 
I left some comments.
please change all classNames into kebab-case, for example instead of FirstName --> first-name. 
class names should not use Capital letters.
Decide between "em"/"px", using them at the same time looks weird unless you have a real reason. 

Please show me a specific Alert for when male > 5 and when female > 5

**Comment by Alex-Fishman** [comment]
also, why   ?!!!?!?!!?!?!?
image

why move buttons have different size?? they should be the same size all the time.

File: src/App.js (hunk 11 above)
Comment (Frickazoid) [reply src/App.js]: Whyshould i remove it? This is for adding some data to page

**Comment by Frickazoid** [comment]
> also, why ?!!!?!?!!?!?!? image
> 
> why move buttons have different size?? they should be the same size all the time.

i'm working on it)

**Pull request merged**